### DISCUSSION TITLE
Force synchronization without data loss - prisma migrate

### DISCUSSION THREAD

**Discussion by Gmulti** [prisma/prisma#7962] [Q&A] [answered]
Hello everyone,

I am in the process of updating prisma to version `2.24.1` and since new features on prisma migrate, I find myself unable to use the migration functions.

Every time, prisma tells me that the database is not synchronized and wants to delete my data: it's just unthinkable.

How can I do it?
I've found a few bits of documentation, especially concerning a production environment, but nothing works.

```
> prisma migrate dev

Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "test", schema "public" at "localhost:5432"

? Drift detected: Your database schema is not in sync with your migration history.

We need to reset the PostgreSQL database "wphealth" at "localhost:5432".
Do you want to continue? All data will be lost.
```

**Comment by ryands17** [comment, marked as answer]
@Gmulti 👋 

What you would need to follow is this guide: [Adding Prisma Migrate to an existing project](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project). 

In this guide, you would need two environments: Development and Production.

1. Development would need to be reset so that you can add migrations via the new Migrate commands
2. Then you need to baseline your production environment to make sure that migrations applied on dev will be in sync. This is done via the `prisma migrate resolve` command as shown in the docs.

Let me know if you have any further queries.

**Reply by ryands17** [reply]
@Gmulti Did the above work for you?

**Reply by Gmulti** [reply]
Sorry @ryands17  I saw the notification but I didn't take the time to reply.

It's all good. I had to resign myself to reset my development database so that it doesn't impact the production but it's "ok".

Do you have any information about shadow databases and the fact that you can migrate like on Heroku? I had some problems on this side but if you have a link on the subject, I'm interested.

**Reply by ryands17** [reply]
Yes, we have docs on that here: https://www.prisma.io/docs/concepts/components/prisma-migrate/shadow-database

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
